lib/ui/painting.dart

```diff
@@ -1580,6 +1713,37 @@ enum ColorSpace {
   /// see the extended values an [ImageByteFormat] like
   /// [ImageByteFormat.rawExtendedRgba128] must be used.
   extendedSRGB,
+  /// The Display P3 color space.
+  ///
+  /// This is a wide gamut color space that has broad hardware support,
+  /// especially Apple devices.
+  ///
+  /// See also: https://en.wikipedia.org/wiki/DCI-P3
+  displayP3,
+}
+
+int _colorSpaceToIndex(ColorSpace colorSpace) {
```

Comment:
Instead make color space an enhanced enum and give it this field?

Also you can do ColorSpace.values[index] for the reverse.

Reply:
I'm not a fan of this because that would put something that is private into the public API.  We could make it inaccessible but it would still show up to people reading through the enum class.

```dart
enum ColorSpace {
  sRGB(0),
  extendedSRGB(1),
  displayP3(2);

  const ColorSpace(this._index);
  final int _index;
}
```

Reply:
That seems ... ok to me? but Its 6 of one, half a. dozen of the other, so either approach is fine to me